File: sampleandroid/src/main/java/com/example/sampleandroid/view/model/ModelScreen.kt
package com.example.sampleandroid.view.model

import com.example.sampleandroid.R
import kim.jeonghyeon.androidlibrary.compose.Screen
import kim.jeonghyeon.androidlibrary.extension.resourceToString

abstract class ModelScreen : Screen() {
    val parentTitle: String = R.string.model.resourceToString()

    companion object {
        val screens = listOf(
            ApiSingleScreen().title to { ApiSingleScreen() },
            ApiSequentialScreen().title to { ApiSequentialScreen() },
            ApiParallelScreen().title to { ApiParallelScreen() },
            ApiPollingScreen().title to { ApiPollingScreen() },
            DbSimpleScreen().title to { DbSimpleScreen() },
            ApiDbScreen().title to { ApiDbScreen() },
            SwitchMapScreen().title to { SwitchMapScreen() }
        )
    }
}

File: sample-native/kotlinIOS/view/model/ModelScreen.swift
//
//  Architecture.swift
//  kotlinIOS
//
//  Created by hyun kim on 18/06/20.
//  Copyright © 2020 hyun kim. All rights reserved.
//

import SwiftUI
import KotlinApi
struct ModelScreen: NavigationScreen {
    var title = "Model".localized()
    @State var model = EmptyViewModelIos()
    
    var content: some View {
        List {
            NavigationLink(destination: ApiSingleScreen()) {
                Text("Api Single Call").font(.headline)
            }
            NavigationLink(destination: ApiSequentialScreen()) {
                Text("Api Sequential Call").font(.headline)
            }
            NavigationLink(destination: ApiParallelScreen()) {
                Text("Api Parallel Call").font(.headline)
            }
            NavigationLink(destination: ApiPollingScreen()) {
                Text("Api Polling").font(.headline)
            }
            NavigationLink(destination: ApiDbScreen()) {
                Text("DB Api Together").font(.headline)
            }
        }
    }
    
}

struct ModelView_Previews: PreviewProvider {
    static var previews: some View {
        ModelScreen()
    }
}
